### ISSUE TITLE
Add multi selection tools in Manager to select multiple features from Map

### ISSUE THREAD

**Issue by azizaparveen** [Esri/solutions-components#772]
### Describe requirements

Right now, Editor widget now only exposes single select tool, but we would like to support multi selection tools like lasso or rectangle in Manager so users can select multiple features from map.

**Comment by azizaparveen** [comment]
cc @sarahmcdonald7

**Comment by sarahmcdonald7** [comment]
@azizaparveen I wonder if we should also consider adding in a way to to select multiple features in the map. We have a selection tool like this in Attachment Viewer and Chart viewer templates. 
![2024-11-27_10-48-49 (1)](https://github.com/user-attachments/assets/f6351f9e-e883-4320-a3cb-47c436893ef9)

**Comment by sarahmcdonald7** [comment]
@azizaparveen I think in the table we should consider updating the app to have the typical selection behavior. So when clicking on multiple check boxes in the table, rather than only selecting one feature at a time multiple features would be selected if you keep checking boxes in the table. If we do this, we should consider adding a Close(X) in the pop-up panel to clear the selection. 

<img width="1061" alt="2024-11-27_10-54-40" src="https://github.com/user-attachments/assets/6ca8b786-4ad6-496a-adb4-921a9ff1eacc">

**Comment by azizaparveen** [comment]
> @azizaparveen I think in the table we should consider updating the app to have the typical selection behavior. So when clicking on multiple check boxes in the table, rather than only selecting one feature at a time multiple features would be selected if you keep checking boxes in the table. If we do this, we should consider adding a Close(X) in the pop-up panel to clear the selection.
> 
@sarahmcdonald7 can we discuss this on Monday? We used to have the typical selection behavior but we changed it because this creates unintended multiple selection where in most cases user just wants to select a single record at a time (#437) , even if we add a close button, we will have to manually click on the close button to unselect it.

**Comment by azizaparveen** [comment]
Tracking change of multi selection behavior of the Table to sync with Map Viewer https://github.com/Esri/solutions-components/issues/1051

**Comment by jmhauck** [comment]
@azizaparveen the associated issue above is closed...do we still have work to do here?

**Comment by azizaparveen** [comment]
@jmhauck yes, we would like to provide multi selection tool to select features from map, like the example Sarah provided in here https://github.com/Esri/solutions-components/issues/772#issuecomment-2504582572.
I think we can push it to R2,
@sarahmcdonald7 what do you guys think?

**Comment by sarahmcdonald7** [comment]
@azizaparveen I think we should be okay to push that to R2. The multi-select tool will primarily be used when we get the batch editing update in R2.

**Comment by sumitzarkar** [comment]
@azizaparveen 

Need more details on this regarding the expected behavior.

**Comment by azizaparveen** [comment]
@sumitzarkar here are some details:

- [ ] Add this option as a map tool toggle 

![Image](https://github.com/user-attachments/assets/2c5dbb17-b5d0-44a1-895c-4bcf0b8d659c)

- [ ] Will show up by default (since we support multi selection in table)
- [ ] If multi selection is off (we would like to add a config option: https://github.com/Esri/solutions-components/issues/1114) it will not show.
- [ ] At runtime, Table count and Information panel will be updated to show multi selection (same behavior would be expected when we have multi selection from table.

**Comment by sumitzarkar** [comment]
@azizaparveen 

 We have analyzed this ticket and have some questions:

## Question 1
 - According to the designs, this tool is meant for selecting drawn graphics, not for selecting features. Therefore, we don’t need to display it for feature selection.
 ![Image](https://github.com/user-attachments/assets/bdb67e16-95eb-4db2-82e9-ab379ee0fb60)
 
- Instead, we can just show the selection tool as  below:
 
  ![Image](https://github.com/user-attachments/assets/b935f114-6644-4458-8cb2-d299abd3ee78)
 
## Question 2
- If some features are already selected from the table or using the map selection tool, what should happen when additional features are selected using the map selection tool?
  - Should the previous selection be cleared, or should the newly selected features be added to the existing selection?

## Question 3
- After drawing a rectangle on the map, should both the drawn rectangle and the highlighted features be displayed, or should the rectangle be cleared once the features are highlighted?
  - If the rectangle remains visible, we may need to provide an additional button to clear the drawn graphic.
 
**Please let us know your thoughts.**

**Comment by azizaparveen** [comment]
@sumitzarkar  

1)  I am good with your suggestion
2) I think it would be good  to have add to selection in this case, it can be hard to select from table because features that need to be selected may not be located adjacent to each other but may be geographically located in close proximity, so in that case  they can select some from table and can select from Map as well. If they only select from map they can always hit Clear selection.
3) I am good to only show selected features.

@sarahmcdonald7 please let us know if you have any concern or suggestion.

**Comment by sarahmcdonald7** [comment]
@azizaparveen I am good with those suggestions.

**Comment by azizaparveen** [comment]
@sumitzarkar this multi selection tool will work this config setting https://github.com/Esri/solutions-components/issues/1114#issuecomment-2786645978

**Comment by pratiksabale07** [comment]
@azizaparveen
 
## Cause
- This is an enhancement requested by the client to show the selection tool in the map.
___
## Solution
![Image](https://github.com/user-attachments/assets/ebdb64c4-a7ed-442d-839b-1b81097c16ca)
- Add a selection tool to the map when selection mode is multiple https://github.com/Esri/solutions-components/issues/1114#issuecomment-2742299684
- Show blue indicator on clicking the tool ( when the tool is active )
- Create a graphics layer to draw a rectangle on the map and then query and retrieve the intersecting features.
- Pass the featuresOIDs to the feature table
- The features will be highlighted on both the map and the table
___
## Dev Checklist
Items  | Is required | Status | Comments
-- | -- | -- | --
Required NLS changes? | Yes   |   | 
Required Backward compatibility | No  |   | 
Required 508 | Yes  |   | 
Need Sanitize? | No  |   | 
___
## Impact Analysis Report
Impacted Areas | Comments
-- | --
Have Configuration changes? | No
Have Runtime changes? | Yes
Impact on RTL | Yes
Impacted components | - Map card</br> - layer table </br> - Info card
Impact on runtime | <ul><li>Selection tool should shown on the map only when selection mode is multiple, for single selection mode it should be hidden</li> <li>When clicked on selection tool a blue dot indicator should be shown at icon (when tool is active)</li><li>On drawing a rectangle the features should be selected from table, info panel and map and drawn rectangle should be cleared</li></ul>

**Comment by azizaparveen** [comment]
Verified in devext
